### ISSUE TITLE
Breaking Change With Cucumber in Version 2.1.11. Need Assistance Please.

### ISSUE THREAD

**Issue by Warren25** [serenity-bdd/serenity-core#2231]
Current Version Serenity Core: 2.3.2
Current Version Serenity Cucumber: 1.9.51

Previous Version Serenity Core: 2.1.10
Previous Version Serenity Cucumber: 1.9.51

Hi John and team,

I updated my serenity core dependency to 2.3.2 and after rebuilding my project, I saw that my cucumber imports were now invalid.

From the release notes, I see there was a Cucumber update with the 2.1.11 release.

I made the necessary updates in step definition files, for example, changing lines like "import cucumber.api.java.en.Then" to "import io.cucumber.java.en.Then".

However, after making all the updates, and attempting to re-run a test in debug mode, I was met with a null pointer exception.

![image](https://user-images.githubusercontent.com/17141304/93384949-7cdb7880-f833-11ea-80f0-8ad24e530453.png)

Is this a Cucumber error? Is there a simple fix?

**Comment by wakaleo** [comment]
You need to use serenity-cucumber6 with serenity-core 2.3.x (see https://github.com/serenity-bdd/serenity-core/blob/master/README.md#what-is-the-latest-stable-version-i-should-use)

**Comment by Warren25** [comment]
@wakaleo 

Thank you so much for the quick response. This seems like an easy fix/update indeed.

I was reading the pom file in the Cucumber 6 starter project here: https://github.com/serenity-bdd/serenity-cucumber-starter/blob/master/pom.xml

And I was confused by line 16, <serenity.cucumber.version>2.3.4</serenity.cucumber.version>

I thought the latest release of serenity cucumber is v1.9.51, noted in this link: https://github.com/serenity-bdd/serenity-cucumber/releases

**Comment by Warren25** [comment]
For reference:

After some moderate reading of several links, I realized I was still using Cucumber 2 with the serenity-cucumber dependency, and needed to use serenity-cucumber6 as @wakaleo mentioned in his previous comment. 

With that known, I removed this line: <serenity-cucumber.version>1.9.51</serenity-cucumber.version>, replaced it with this line: <serenity.cucumber.version>2.3.4</serenity-cucumber.version>

I also removed this dependency: 
<dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-cucumber</artifactId>
            <version>${serenity-cucumber.version}</version>
        </dependency>

And replaced with this dependency: 
<dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-cucumber6</artifactId>
            <version>${serenity.cucumber.version}</version>
            <scope>test</scope>
        </dependency>


And then finally, after attempting to run a test in de-bug mode, I ran into one final issue, apparently an IntelliJ issue, which I found the solution here: https://github.com/serenity-bdd/serenity-cucumber4/issues/20

Everything is working fine now, as it was.

This issue can be closed.
